### PR TITLE
Ensure that sim_joints_ always has the same number of elements as the…

### PR DESCRIPTION
**Pull request by kbogert**
… joint_names

Otherwise a segfault occurs when some joints are not simulated by gazebo.

Possbily the cause of #67

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
I'm not sure about adding a nullptr to the vector. This is happening because the joint_name doesn't exist, maybe it's because your configuration is wrong? or is there any specific use case for this ?

**Comment by kbogert** [comment]
This is happening because there are joints in the urdf file that are not simulated by gazebo, there are lots of reasons why this could be the case (in our case, it's due to hacks needed to get parallel grippers to move correctly).

The alternative is to remove these joints from all of the vectors, reducing their size as the non-simulated joints are discovered.  This is do-able but much more complicated, and as far as I can tell having the nulls in the sim_joints vector isn't causing any issues.

File: gazebo_ros2_control/src/gazebo_system.cpp [hunk 1]
```diff
@@ -283,6 +298,9 @@ hardware_interface::return_type GazeboSystem::write()
   rclcpp::Duration sim_period = sim_time_ros - this->dataPtr->last_update_sim_time_ros_;
 
   for (unsigned int j = 0; j < this->dataPtr->joint_names_.size(); j++) {
+    if (this->dataPtr->sim_joints_[j]) {
```
Comment (ahcorde) [inline gazebo_ros2_control/src/gazebo_system.cpp:301]: This indentation looks wrong

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: gazebo_ros2_control/src/gazebo_system.cpp [hunk 2]
```diff
@@ -283,20 +298,23 @@ hardware_interface::return_type GazeboSystem::write()
...
-      this->dataPtr->sim_joints_[j]->SetVelocity(
-        0,
-        this->dataPtr->joint_velocity_cmd_[j]);
-    }
-    if (this->dataPtr->joint_control_methods_[j] & EFFORT) {
-      const double effort =
-        this->dataPtr->joint_effort_cmd_[j];
-      this->dataPtr->sim_joints_[j]->SetForce(0, effort);
+    if (this->dataPtr->sim_joints_[j]) {
+
```
Comment (ahcorde) [inline gazebo_ros2_control/src/gazebo_system.cpp:302]: remove this line, then it's good to go

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Review by ahcorde — CHANGES_REQUESTED** [review]
@kbogert  uncrustify is failing

**Review by ahcorde — APPROVED** [review]

**Pull request merged**